spec.md

```diff
@@ -3873,27 +3873,51 @@ id = ( "_" | letter ) , { "_" | letter | digit_dec } | literal_id ;
 
 (* Fileinfo communicates Chisel source file and line/column info *)
 linecol = digit_dec , { digit_dec } , ":" , digit_dec , { digit_dec } ;
-lineinfo = string, " ", linecol
+lineinfo = string, " ", linecol ;
 info = "@" , "[" , lineinfo, { ",", lineinfo }, "]" ;
 
 (* Type definitions *)
+type = ( [ "const" ] , type_hardware ) | type_probe ;
+
+type_hardware =
+    type_ground
+  | type_bundle
+  | type_vec
+  | type_enum
+  | id ;
+
+(* Ground Types *)
+type_ground =  type_ground_nowidth | type_ground_width ;
+
+type_ground_nowidth =
+    "Clock"
+  | "Reset"
+  | "AsyncReset" ;
+
+type_ground_width =
+    "UInt" , [ width ]
+  | "SInt" , [ width ]
+  | "Analog" , [ width ] ;
+
 width = "<" , int , ">" ;
-type_ground = "Clock" | "Reset" | "AsyncReset"
-            | ( "UInt" | "SInt" | "Analog" ) , [ width ] ;
-type_enum = "{|" , { field_enum } , "|}" ;
-field_enum = id, [ ":" , type_simple_child ] ;
-type_aggregate = "{" , field , { field } , "}"
-               | type , "[" , int , "]" ;
-type_ref = ( "Probe" | "RWProbe" ) , "<", type , [ "," , id , "," ] ">" ;
-field = [ "flip" ] , id , ":" , type ;
-type_property = "Integer" ;
-type_simple_child = type_ground | type_enum | type_aggregate | id ;
-type = ( [ "const" ] , type_simple_child ) | type_ref ;
 
-(* Type alias declaration *)
-type_alias_decl = "type", id, "=", type ;
+(* Bundle Types *)
+type_bundle = "{" , type_bundle_field , { type_bundle_field } , "}" ;
+type_bundle_field = [ "flip" ] , id , ":" , type ;
+
+(* Vec Types *)
+type_vec = type , "[" , int , "]" ;
+
+(* Enum Types *)
+type_enum = "{|" , { type_enum_alt } , "|}" ;
+type_enum_alt = id, [ ":" , type_constable ] ;
```

Comment:
It's certainly reasonable in programming languages. It would amount to an empty type.

I'll put a note on my cleanup list to talk about this potentiality. Thanks for pointing this out.